libraries/AP_Motors/AP_MotorsMatrix.h

```diff
@@ -90,14 +90,33 @@ class AP_MotorsMatrix : public AP_MotorsMulticopter {
     // the arguments to add_motor - with the exception of the type,
     // for compactness.
     struct MotorDefInt {
-        int8_t motor_num;
         int16_t angle_degrees;
         int8_t yaw_factor;
         uint8_t testing_order;
     };
+    // same structure, but with floats.
+    struct MotorDef {
+        float angle_degrees;
+        float yaw_factor;
+        uint8_t testing_order;
+    };
 
     // method to add many motors specified in a structure:
-    void add_motors(const struct MotorDefInt *motor, uint8_t num_motors);
+    template <typename T>
+    void add_motors(T *motor, uint8_t num_motors);
+
+    // structure used for initialising motors that add have separate
+    // roll/pitch/throttle factors.  Note that this does *not* include
```

Comment:
```suggestion
    // roll/pitch/yaw throttle factors.  Note that this does *not* include
```

Reply:
Fixed in PR this is based on, and this one rebased.